### DISCUSSION TITLE
Receiving different types from Prisma after updating to 4.16.1

### DISCUSSION THREAD

**Discussion by leosavvy** [prisma/prisma#20009] [Q&A] [answered]
Hello,

I've been using Prisma on a NestJS project, this is the Prisma Wrapper I use for Injecting the prisma instance on my modules:

```
@Injectable()
export class Prisma extends PrismaClient implements OnModuleInit {
  async onModuleInit(): Promise<boolean> {
    await this.$connect();
    return true;
  }

  async enableShutdownHooks(app: INestApplication): Promise<void> {
    this.$on('beforeExit', async () => {
      await this.$disconnect();
      await app.close();
    });
  }
}
```

Before updating to 4.16.1 the types Prisma generated were like:

```
(alias) type AppConfig = {
    id: number;
    frontEndURL: string;
    pythonURL: string;
}
import AppConfig
```

After updating I'm now receiving types like:

```
(alias) type AppConfig = runtime.GetResult<{
    id: number;
    frontEndURL: string;
    pythonURL: string;
}, unknown> & {}
import AppConfig
```

This is causing errors on all other layers of my app. 

On this discussion: https://github.com/prisma/prisma/discussions/18216#discussioncomment-6315363 it's mentioned this is caused by Client Extensions, I don't remember ever enabling them and can't find a way of permanently disable this new feature.

Current Versions I'm using:

```
"prisma": "^4.16.1"
"@prisma/client": "^4.16.1"
```

**Comment by nurul3101** [comment]
Hello @leosavvy 👋 

Prisma Client Extensions are available in GA with version 4.16.0 and above.
Does the error go away if you use version 4.15.0?
Are you using `prisma.$extends` ([reference](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#extends)) anywhere in your code base?

**Reply by nurul3101** [reply]
@leosavvy I am checking to see if you had a chance to look at my last response.

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by leosavvy** [reply]
Hello @nurul3101, sorry for the late response!

I'm not using prisma.$extends anywhere in my code:
![image](https://github.com/prisma/prisma/assets/22433902/f725fc5d-0c1b-4e3d-bbe0-ff59c355bbf1)

And although I tried multiple times to downgrade the version of my prisma client, both locally on the project and globally on my system it's still generating the client with the latest version.

I've found a way to bypass the errors I was having and work along this Prisma behavior so I'm gonna close this discussion.

Thank you very much for your answers!

**Reply by MineiToshio** [reply]
@leosavvy how did you solve this? I'm having the same problem.

**Reply by leosavvy** [reply]
I did not solve it, I closed the discussion assuming this was the new 'normal'

**Reply by janpio** [reply]
Is this the same as https://github.com/prisma/prisma/issues/20320?
If so, that will be changed in Prisma 5.1.0 which will be released beginning of August.
You can already try `prisma@dev` and `@prisma/client@dev` to test the new behavior.
Let us know if this improves the situation for you.

**Comment by SimHoZebs** [comment]
I'm experiencing the same issue on the latest version (4.16.2). But unlike the example given here, I get the `GetResult<{...}>` type on all of my prismaClient queries instead.

**Reply by MineiToshio** [reply]
Did you fix it @SimHoZebs?

**Reply by SimHoZebs** [reply]
Nope!

**Reply by benjibenoit** [reply]
Same problem for me !

**Reply by janpio** [reply]
Is this the same as https://github.com/prisma/prisma/issues/20320?
If so, that will be changed in Prisma 5.1.0 which will be released beginning of August.
You can already try `prisma@dev` and `@prisma/client@dev` to test the new behavior.
Let us know if this improves the situation for you.

**Reply by benjibenoit** [reply]
> Is this the same as #20320? If so, that will be changed in Prisma 5.1.0 which will be released beginning of August. You can already try `prisma@dev` and `@prisma/client@dev` to test the new behavior. Let us know if this improves the situation for you.

For me yes ! I'll test

**Reply by SimHoZebs** [reply]
> Is this the same as #20320? If so, that will be changed in Prisma 5.1.0 which will be released beginning of August. You can already try `prisma@dev` and `@prisma/client@dev` to test the new behavior. Let us know if this improves the situation for you.

I installed both dev packages but do not see a change.

**Reply by benjibenoit** [reply]
> > Is this the same as #20320? If so, that will be changed in Prisma 5.1.0 which will be released beginning of August. You can already try `prisma@dev` and `@prisma/client@dev` to test the new behavior. Let us know if this improves the situation for you.
> 
> I installed both dev packages but do not see a change.

Yep, same for me. I still have ```runtime.GetResult<>``` 🫤

**Reply by janpio** [reply]
Can you share your output of `npx prisma -v`? This should be available starting with `5.1.0-dev.60`.

**Reply by benjibenoit** [reply]
> Can you share your output of `npx prisma -v`? This should be available starting with `5.1.0-dev.60`.
```
prisma                  : 5.1.0-dev.61
@prisma/client          : 5.1.0-dev.61
Current platform        : debian-openssl-3.0.x
Query Engine (Node-API) : libquery-engine 4a1c72356ffb884a5752157c21d8bac67afd1720 (at ../../node_modules/@prisma/engines/libquery_engine-debian-openssl-3.0.x.so.node)
Schema Engine           : schema-engine-cli 4a1c72356ffb884a5752157c21d8bac67afd1720 (at ../../node_modules/@prisma/engines/schema-engine-debian-openssl-3.0.x)
Schema Wasm             : @prisma/prisma-schema-wasm 5.1.0-9.4a1c72356ffb884a5752157c21d8bac67afd1720
Default Engines Hash    : 4a1c72356ffb884a5752157c21d8bac67afd1720
Studio                  : 0.487.0
```

**Reply by Jolg42** [reply]
@shankiflang Did you run `npx prisma generate`?
Also, after running generate you could try restarting your IDE & doing a restart of the TS Server?
If that still does not work, could you 
- give an example code of a query where this is happening?
- tell us which IDE you are using here?

**Reply by benjibenoit** [reply]
> @shankiflang Did you run `npx prisma generate`? 

Yes, I ran ```yarn worksace XXX pnpify prisma generate``` as I'm in a workspace yarn2

> Also, after running generate you could try restarting your IDE & doing a restart of the TS Server?

Yes, i'm working with docker.

> give an example code of a query where this is happening?

Mmmmh. In fact, everything seems to be working. It generates everything well under a runtime.GetResult that I wasn't used to seeing, but the error I had was a mistake on my part.

**Reply by SimHoZebs** [reply]
@Jolg42 I wish I could provide a codesandbox copy of my issue, but prisma types don't seem to show up as anything more than `any` when I share it publicly even after running `prisma generate`. Below is a snippet instead, but you can see the copy [here](https://codesandbox.io/p/github/SimHoZebs/nedon/draft/divine-dream?file=%2Fpackage.json%3A7%2C19&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522vertical%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522clkgw0q2x046o3o6qh3qoy517%2522%252C%2522sizes%2522%253A%255B64.4295814645802%252C35.570418535419805%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clkgw0q2w046l3o6qdnd7mpc4%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clkgw0q2w046m3o6qt0rx0yl6%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clkgw0q2w046n3o6qvnax31hv%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B56.51960784%252C43.48039216%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clkgw0q2w046l3o6qdnd7mpc4%2522%253A%257B%2522id%2522%253A%2522clkgw0q2w046l3o6qdnd7mpc4%2522%252C%2522activeTabId%2522%253A%2522clkgx9f9102v73o6us2puwwwg%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522id%2522%253A%2522clkgx76gf02363o6uo8bxbyjf%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fserver%252Frouters%252Fsplit.ts%2522%252C%2522id%2522%253A%2522clkgx9f9102v73o6us2puwwwg%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522clkgw0q2w046m3o6qt0rx0yl6%2522%253A%257B%2522id%2522%253A%2522clkgw0q2w046m3o6qt0rx0yl6%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clkgw0q2w046n3o6qvnax31hv%2522%253A%257B%2522id%2522%253A%2522clkgw0q2w046n3o6qvnax31hv%2522%252C%2522activeTabId%2522%253A%2522clkgx6rwx01ui3o6uymq1quyu%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522build%2522%252C%2522id%2522%253A%2522clkgx013c00m73o6uro77whch%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522id%2522%253A%2522clkgx6rwx01ui3o6uymq1quyu%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clkgx6ski0034gxhu0jd2d0rj%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D).

```ts
// server/router/split.ts
import db from "./db"

const test = async () => {
  const split = await db.split.create({
    data: //split data here
  })
   /*split
    const split: GetResult<{
    id: string;
    transactionId: string;
    userId: string;
}, unknown> & {}*/
}
```

<details> 

<summary>db.ts</summary>

```ts
import { PrismaClient } from "@prisma/client";

declare global {
  var prisma: PrismaClient | undefined;
}

const prisma = global.prisma || new PrismaClient();

if (process.env.NODE_ENV !== "production") global.prisma = prisma;

export default prisma;
```

</details>

<details> 

<summary>prisma -v</summary>

```
prisma                  : 5.1.0-dev.63
@prisma/client          : 5.1.0-dev.63
Current platform        : windows
Query Engine (Node-API) : libquery-engine 6a4762d4d94d9c5aa24faf7803734c9af7fa014f (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Schema Engine           : schema-engine-cli 6a4762d4d94d9c5aa24faf7803734c9af7fa014f (at node_modules\@prisma\engines\schema-engine-windows.exe)
Schema Wasm             : @prisma/prisma-schema-wasm 5.1.0-13.6a4762d4d94d9c5aa24faf7803734c9af7fa014f
Default Engines Hash    : 6a4762d4d94d9c5aa24faf7803734c9af7fa014f
Studio                  : 0.487.0
```
</details>

I use VSCode with typescript version 5.1.6.

**Reply by Jolg42** [reply]
@SimHoZebs Did you run `yarn prisma generate`? For me that did the trick, the types needed to be generated.

**Reply by SimHoZebs** [reply]
> @SimHoZebs Did you run `yarn prisma generate`? For me that did the trick, the types needed to be generated.

I did :( I restarted TS server and the IDE as you instructed, and also tried removing `zod-prisma` package, node_modules, and yarn.lock and reinstalled everything before generating again. There must be something about my project that's screwing it up, so I'll update once I can reproduce it in a different environment.

**Reply by normelton** [reply]
I may be having a similar problem as @SimHoZebs. I've installed prisma/prisma-client 5.1.0-dev.77, completely blow away/reinstalled my node_modules directory, and regenerated my types. But still, a query result is getting tangled up in a `Prisma.Prisma__UserClient<GetResult_2<>>` construct. See the below screenshot.

<img width="451" alt="Screenshot 2023-07-27 at 4 03 25 PM" src="https://github.com/prisma/prisma/assets/30230/7c52e461-5d9a-4c59-b941-463e76e73f36">

**Reply by Jolg42** [reply]
Could you try with `5.1.0`? This PR improved this https://github.com/prisma/prisma/pull/20435

**Reply by SimHoZebs** [reply, marked as answer]
`5.1.1` solved the issue!

**Reply by normelton** [reply]
Same, thanks!
